src/mock_display.rs

```diff
@@ -563,12 +633,195 @@ where
+        if bounding_box.contains(point_top) {
+            let color = display
+                .get_pixel(point_top)
+                .map(|c| c.into())
+                .unwrap_or(C::NONE_COLOR);
+
+            write!(f, "\x1b[38;2;{};{};{}m", color.r(), color.g(), color.b())?;
+        } else {
+            write!(f, "\x1b[39m")?;
+        };
+
+        // Set background color.
+        if bounding_box.contains(point_bottom) {
+            let color = display
+                .get_pixel(point_bottom)
+                .map(|c| c.into())
+                .unwrap_or(C::NONE_COLOR);
+
+            write!(f, "\x1b[48;2;{};{};{}m", color.r(), color.g(), color.b())?;
+        } else {
+            write!(f, "\x1b[49m")?;
+        };
+
+        // Write "upper half block" character.
+        f.write_char('\u{2580}')?;
+    }
+
+    // Reset colors.
+    f.write_str("\x1b[0;m")?;
+
+    // Pad output with spaces if column width is larger than the width of the bounding box.
+    for _ in bounding_box.size.width as usize..column_width {
+        f.write_char(' ')?
+    }
+
+    Ok(())
+}
+
+fn write_vertical_border(f: &mut fmt::Formatter<'_>, column_width: usize) -> fmt::Result {
+    write!(
+        f,
+        "+-{:-<width$}-+-{:-<width$}-+-{:-<width$}-+\n",
+        "",
+        "",
+        "",
+        width = column_width
+    )
+}
+
+fn write_header(f: &mut fmt::Formatter<'_>, column_width: usize) -> fmt::Result {
+    write_vertical_border(f, column_width)?;
+    write!(
+        f,
+        "| {:^width$} | {:^width$} | {:^width$} |\n",
+        "display",
+        "expected",
+        "diff",
+        width = column_width
+    )?;
+    write_vertical_border(f, column_width)
+}
+
+impl<C> Display for FancyPanic<'_, C>
+where
+    C: PixelColor + ColorMapping,
+{
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        let diff = self.display.diff(self.expected);
+
+        let bounding_box_display = self.display.affected_area_origin();
+        let bounding_box_expected = self.expected.affected_area_origin();
+
+        let bounding_box = Rectangle::new(
+            Point::zero(),
+            bounding_box_display
+                .size
+                .component_max(bounding_box_expected.size),
+        );
+
+        f.write_char('\n')?;
+
+        // Output the 3 displays in columns if they are less than 30 pixels wide.
+        if bounding_box.size.width <= 30 {
+            // Set the width of the output columns to the width of the bounding box,
+            // but at least 10 characters to ensure the column labels fit.
+            let column_width = (bounding_box.size.width as usize).max(10);
+
+            write_header(f, column_width)?;
+
+            for y in bounding_box.rows() {
+                if y % 2 != 0 {
+                    // Skip all odd y coordinates, because `write_row` outputs two pixel rows with
+                    // a single row of characters.
+                    continue;
+                }
```

Comment:
Same change here I think?

```suggestion
            for y in bounding_box.rows().step_by(2) {
```